### DISCUSSION TITLE
Is it possible to make this work behind nginx proxy?

### DISCUSSION THREAD

**Discussion by YannisMarios** [thgh/rollup-plugin-livereload#79] [Q&A]
Hi, 

I am trying to make reload work behind nginx proxy. Is this possible? 

right now my nginx config has:

```
...
location / {
  proxy_pass http://localhost:5000;
}

listen 443 ssl;
...
```

Thank you.

**Comment by davidmroth** [comment]
Super easy! Here are the relevant components needed to make this run behind an nginx proxy...

**rollup.config.js**
```
import serve from 'rollup-plugin-serve'
import livereload from 'rollup-plugin-livereload';

...
...


export default {

...,

plugins: [
...,

livereload({
    watch: '<folder you want to watch for changes>',
    verbose: true,
    delay: 200,
}),

...,

serve({
      verbose: true,
      contentBase: ['public'],
      historyApiFallback: true,
      host: '0.0.0.0',
      port: 5000,
    }),
]
```

**nginx config:**
 ```
server {
  listen 80;
    location / {
      proxy_pass  http://<hostname where your node app is being served>:5000/;
      proxy_redirect  default;
      proxy_set_header  Upgrade $http_upgrade;
      proxy_set_header  Connection "upgrade";
      proxy_set_header  Host $host;
      proxy_set_header  X-Real-IP $remote_addr;
      proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header  X-Forwarded-Host $server_name;
      proxy_set_header  X-Forwarded-Proto $scheme;
    }
}

server {
  listen 35729;

  location / {
    proxy_pass http://<hostname of where your node app lives>:35729/;
    proxy_redirect off;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-NginX-Proxy false;
    proxy_http_version 1.1;
    # prevents 502 bad gateway error
    proxy_buffers 8 32k;
    proxy_buffer_size 64k;
  }
}
```

_**Note:** The last part is for the websocket. Also, pay special attention to the host and port (0.0.0.0 and 5000 respectively). Lastly, keep in mind that rollup-plugin-livereload loads a module that listens on port 35729 for the changes (using a websocket)_

**Comment by enf-alan** [comment]
Hello, is their a way to make to live reload laravel, e.g. my-app.test domain?

**Comment by davidmroth** [comment]
> Hello, is their a way to make to live reload laravel, e.g. my-app.test domain?

Not sure what you're asking. Can you provide more details?

**Comment by jdoubleu** [comment]
@YannisMarios did [this](https://github.com/thgh/rollup-plugin-livereload/issues/58#issuecomment-782906653) help you?

> Hello, is their a way to make to live reload laravel, e.g. my-app.test domain?
I think you are looking for a way to include the bundle from a different host. 

You could use the `options.clientUrl` to hard-code the url to your local webserver:
```js
  // ...
  livereload({
    watch: your_path,
    clientUrl: 'http://localhost:35729/livereload.js?snipver=1'
  }),
  // ...
```

My PR #72 adds an additional option, which makes this easier to use, i.e. you do not have to hard-code the absolute URL.

**Comment by jonkoops** [comment]
I'm converting this issue to a discussion, as it's more of a question than a bug or a feature request.
